docs/system_overview/genquery.md

```diff
@@ -553,3 +551,331 @@ Find all data objects with replicas on resource `replResc` which are not marked
+  [
+    "job"
+  ],
+  [
+    ""    # <------ Look suspicious?
+  ]
+]
+```
+
+Here's the SQL that produced the resultset.
+
+```sql
+SELECT DISTINCT
+    mmc.meta_attr_name
+FROM
+    R_COLL_MAIN t0
+    INNER JOIN R_OBJT_ACCESS pcoa ON t0.coll_id = pcoa.object_id
+    INNER JOIN R_TOKN_MAIN pct ON pcoa.access_type_id = pct.token_id
+    INNER JOIN R_USER_MAIN pcu ON pcoa.user_id = pcu.user_id
+    LEFT JOIN R_OBJT_METAMAP ommc ON t0.coll_id = ommc.object_id
+    LEFT JOIN R_META_MAIN mmc ON ommc.meta_id = mmc.meta_id
+WHERE
+    pcu.user_name = ?
+    AND pcu.zone_name = 'tempZone'
+    AND pcoa.access_type_id >= 1050 FETCH FIRST 256 ROWS ONLY
+```
+
+Based on the generated SQL, the resultset returned by GenQuery2 is correct. The SQL produces empty strings for ALL collections that do NOT have metadata attached to them.
+
+To get around this, apply an additional condition to the query like so.
+
+```sh
+$ iquery "select META_COLL_ATTR_NAME where META_COLL_ATTR_NAME is not null" | jq
+[
+  [
+    "bob"
+  ],
+  [
+    "job"
+  ]
+]
+```
+
+### Sorting
+
+Sorting data with GenQuery2 uses the same syntax as standard SQL. For example:
+
+```sh
+iquery "select RESC_NAME, COLL_NAME, DATA_NAME order by RESC_NAME desc, COLL_NAME, DATA_NAME asc"
+```
+
+Notice the use of `desc` and `asc`. Just like standard SQL, users can specify the sorting direction.
+
+- Ascending order is specified by passing `asc` and is the _implicit_ default
+- Descending order is specified by passing `desc`
+
+Any number of attributes can be passed to the ORDER-BY clause as long as those attributes are defined in the PROJECTION clause.
+
+### Escaping Bytes and Special Characters
+
+Sometimes you may need to escape a specific byte (or character). GenQuery2 supports this through the use of hexadecimal encoding.
+
+!!! Note
+    This only applies to **string literals**. That is, the arguments wrapped in single quotes. Only **one byte** can be encoded at a time.
+
+This is achieved by using `\xNN`, where _NN_ is the hexadecimal value of the byte. GenQuery2 decodes the byte before it reaches the database.
+
+The following example demonstrates how to escape an exclamation mark.
+
+```sh
+iquery "select COLL_NAME, DATA_NAME where DATA_NAME = 'data\x21.txt'"
+```
+
+If the catalog contains a data object by the name, `data!.txt`, then it will be returned by GenQuery2.
+
+Embedded single quotes can be escaped by adding another single quote, just like standard SQL. For example:
+
+```sh
+iquery "select COLL_NAME, DATA_NAME where DATA_NAME = 'that''s my data.txt'"
+```
+
+GenQuery2 will notice the use of double single quotes and collapse them to one single quote before sending to the database. In the case of the example, that means `that''s` will be passed to the database as `that's`.
+
+### Showing Duplicate Entries
+
+By default, GenQuery2 removes duplicate entries from the resultset.
+
+If you need to view duplicate entries, add `no distinct` after the `select` keyword. For example:
```

Comment:
is this standard SQL?  is there a better syntax for this?

Reply:
`no distinct` is not support by the SQL standard.

I pulled that forward during the early development stages of the parser and never went back to change it. Finding a better way to express that is on my TODO list. One way we could address is by doing something like what's described here:  https://github.com/irods/irods_api_plugin_genquery2/issues/42#issuecomment-1712525639. Alternatively, we introduce more support for the distinct keyword so that it behaves as defined by the SQL standard. I believe SQL allows users to apply the distinct keyword to one or more columns. Or maybe that's database specific. I'll verify and open an issue for it.

Reply:
right, sticking to standard SQL is definitely the answer if plausible.  thanks.     please resolve this once you open a new issue.

Reply:
Created irods/irods#7944.

Reply:
sold